src/Core_Fix_InvalidSceneFileProtection/InvalidSceneFileProtection.cs

```diff
@@ -41,6 +43,64 @@ private static bool OnClickImportPrefix(List<string> ___listPath, int ___select)
             return IsFileValid(path);
         }
 
+        class Chunk
+        {
+            public static int Size = 1 << 20;
+            public byte[] bytes = new byte[Size];
+            public int readed;
+        }
+
+        class SearchStatus
+        {
+            public volatile bool found;
+        }
+
+        class PoolAllocator<T> where T : new()
+        {
+            private Queue<T> _queue = new Queue<T>();
+            private volatile int _remaingNewCount;
+            private int _maxPool;
+
+            public PoolAllocator( int maxPool )
+            {
+                _maxPool = maxPool;
+                _remaingNewCount = maxPool;
+            }
+
+            public T Acquire()
+            {
+                lock (_queue)
+                {
+                    while (_queue.Count <= 0 && _remaingNewCount <= 0)
+                        Monitor.Wait(_queue);
```

Comment:
Maybe a timeout would be a good idea just in case? I think there's no ill effect in normal conditions.
```suggestion
                        Monitor.Wait(_queue, 2000);
```

Reply:
I added it just in case.